tests/components/manual/test_alarm_control_panel.py

```diff
@@ -421,6 +422,130 @@ async def test_arm_night_with_invalid_code(hass):
     assert hass.states.get(entity_id).state == STATE_ALARM_DISARMED
 
 
+async def test_arm_vacation_no_pending(hass):
```

Comment:
Instead of duplicating the tests, consider parametrizing them, maybe something like this:

```
@pytest.mark.parametrize(
    "service,expected_state",
    [
        (SERVICE_ALARM_ARM_VACATION, STATE_ALARM_ARMED_VACATION),
        ...
    ],
)
async def test_no_pending(hass, service, expected_state):
    """Test no pending after arming."""
    assert await async_setup_component(
        hass,
        alarm_control_panel.DOMAIN,
        {
            "alarm_control_panel": {
                "platform": "manual",
                "name": "test",
                "code": CODE,
                "arming_time": 0,
                "disarm_after_trigger": False,
            }
        },
    )
    await hass.async_block_till_done()

    entity_id = "alarm_control_panel.test"

    assert hass.states.get(entity_id).state == STATE_ALARM_DISARMED

    await hass.services.async_call(
        alarm_control_panel.DOMAIN,
        service,
        {ATTR_ENTITY_ID: "alarm_control_panel.test", ATTR_CODE: CODE},
        blocking=True,
    )

    assert hass.states.get(entity_id).state == expected_state
```

Reply:
I will try to do this

Reply:
Done